test/sensor.js

```diff
@@ -8,7 +8,43 @@ var MockFirmata = require("./util/mock-firmata"),
     io: new MockFirmata(),
     debug: false,
     repl: false
-  });
+  }),
+  // TODO: query: does node.js / es6 have an object clone / deepcopy method?
```

Comment:
Currently only using the deep copy in the sensor test code.  When I get to adding the property to provide a readonly copy of the (to be) updated samples array, I'll be wanting to use it there as well.  Seems the simplest way to create a copy of an array of objects.